src/main/webapp/script.js

```diff
@@ -149,23 +163,15 @@ function noPotentialMatch() {
 }
 
 function matchButtonPressed() {
-  const decision = {
-      userid: getCurrentUserId(),
-      potentialMatchID: currentPMDisplayed,
-      decision: FRIENDED
-  }
-  fetch('/match-decisions', { method: 'POST', body: JSON.stringify(decision) }).then((response) => {
+  const userid = getCurrentUserId();
+  fetch('/match-decisions?userid=' + userid + "&potentialMatchID=" + currentPMDisplayed + "&decision=" + FRIENDED, { method: 'POST' }).then((response) => {
```

Comment:
nit: this line is pretty long, consider splitting or wrapping it for readability